test/unit/cloudinaryUtils.test.js

```diff
@@ -21,35 +21,33 @@ const tracks = [
     srclang: 'se'
   }
 ];
-import Utils from '../../src/utils';
-jest.doMock('fetch-ponyfill/build/fetch-browser', () => ({
-  __esModule: true,
-  fetch: jest.fn((url, options) => {
-    if (url === tracks[2].src) {
-      return Promise.resolve({
-        json: () => Promise.resolve({ status: 404 })
+jest.mock('fetch-ponyfill/build/fetch-browser', () => (() => {
+  return {
+    fetch: jest.fn((url, options) => {
+      return new Promise((resolve) => {
+        if (url === tracks[2].src) {
+          resolve({ status: 404 });
+        } else {
+          resolve({ status: 200 });
+        }
       });
-    } else {
-      Promise.resolve({
-        json: () => Promise.resolve({ status: 200 })
-      });
-    }
-  })
+    }),
+    default: jest.fn()
+  };
 }));
-import fetchPF from 'fetch-ponyfill/build/fetch-browser';
-
-// require('jest-fetch-mock').enableMocks();
+import { fetch } from 'fetch-ponyfill/build/fetch-browser';
+import { filterAndAddTextTracks } from '../../src/utils/cloudinary';
 
 describe('video source tests', () => {
-  it('test filter out bad vtt', async () => {
-    jest.useFakeTimers();
+  it('test filter out bad vtt', async (done) => {
     let vjs = jest.createMockFromModule('video.js');
     vjs.addRemoteTextTrack = jest.fn();
     jest.spyOn(vjs, 'addRemoteTextTrack');
-    let a = await Utils.filterAndAddTextTracks(tracks, vjs);
-    console.log(a);
-    // expect(vjs.addRemoteTextTrack).toBeCalled();
-
+    filterAndAddTextTracks(tracks, vjs);
```

Comment:
Looks like 'filterAndAddTextTracks' is a function that make async action. 
It used fetch inside a forEach. But the function does not return anything.  
Using a  'setTimeout' is not a solution.  
Please make the function into a pure function.  it should  return a Promise 
 and resolve it when all the fetching are done.

Reply:
Or using Promise.All and call ' videojs.addRemoteTextTrack' after all fetch is done

Reply:
@e1adn I don't what to change working code just to satisfy a test method.
It is like that because these are basically "async workers" they do something we do not need return, as it would block the flow for no good reason, we use the work result in a later stage.  they build the result independent of each other.